File: client/Core/API/FileSystem.cs
using Godot;
using IronRuby.Builtins;

namespace API
{
    public class FileSystem
    {
        public FileSystem()
        {

        }

        public bool exists(string path)
        {
            var file = new File();
            return file.FileExists(path);
        }

        public void write_all_text(string path, string text)
        {

        }

        public void write_all_bytes(string path, byte[] buff)
        {

        }

        public string read_all_text(string path)
        {
            var file = new File();
            file.Open(path, File.ModeFlags.Read);
            var content = file.GetAsText();
            file.Close();
            return content;
        }

        public MutableString read_all_bytes(string path)
        {
            var file = new File();
            file.Open(path, File.ModeFlags.Read);

            var len = (int)file.GetLen();
            var content = file.GetBuffer(len);
            var buff = MutableString.CreateBinary(content);

            file.Close();
            return buff;
        }
    }
}

File: client/Scripts/file_system.rb
module FileSystem
  def self.init(api)
    @@api = api
  end

  def self.exists(path)
    @@api.exists path
  end

  def self.read_all_text(path)
    @@api.read_all_text path
  end

  def self.read_all_bytes(path)
    @@api.read_all_bytes path
  end

  def self.write_all_text(path, text)
    @@api.write_all_text path, text
  end

  def self.write_all_bytes(path, buff)
    @@api.write_all_bytes path, buff
  end
end
